domain/model/state/modelstate.go

```diff
@@ -106,12 +112,220 @@ func (s *ModelState) Delete(ctx context.Context, uuid coremodel.UUID) error {
 	return nil
 }
 
+func (s *ModelState) getModelUUID(ctx context.Context, tx *sqlair.TX) (coremodel.UUID, error) {
+	var modelUUID dbUUID
+	stmt, err := s.Prepare(`SELECT &dbUUID.uuid FROM model;`, dbUUID{})
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	err = tx.Query(ctx, stmt).Get(&modelUUID)
+	if errors.Is(err, sql.ErrNoRows) {
+		return "", errors.New("model does not exist").Add(modelerrors.NotFound)
+	}
+	if err != nil {
+		return "", errors.Errorf("getting model uuid: %w", err)
+	}
+
+	return coremodel.UUID(modelUUID.UUID), nil
+}
+
 // GetModelConstraints returns the currently set constraints for the model.
 // The following error types can be expected:
 // - [modelerrors.NotFound]: when no model exists to set constraints for.
+// - [modelerrors.ConstraintsNotFound]: when no model constraints have been
+// set for the model.
 func (s *ModelState) GetModelConstraints(ctx context.Context) (constraints.Value, error) {
-	// TODO: Implement this method.
-	return constraints.Value{}, nil
+	db, err := s.DB()
+	if err != nil {
+		return constraints.Value{}, errors.Capture(err)
+	}
+
+	selectTagStmt, err := s.Prepare(
+		"SELECT &dbConstraintTag.* FROM v_model_constraint_tag", dbConstraintTag{},
+	)
+	if err != nil {
+		return constraints.Value{}, errors.Capture(err)
+	}
+
+	selectSpaceStmt, err := s.Prepare(
+		"SELECT &dbConstraintSpace.* FROM v_model_constraint_space", dbConstraintSpace{},
+	)
+	if err != nil {
+		return constraints.Value{}, errors.Capture(err)
+	}
+
+	selectZoneStmt, err := s.Prepare(
+		"SELECT &dbConstraintZone.* FROM v_model_constraint_zone", dbConstraintZone{})
+	if err != nil {
+		return constraints.Value{}, errors.Capture(err)
+	}
+
+	var (
+		cons   dbConstraint
+		tags   []dbConstraintTag
+		spaces []dbConstraintSpace
+		zones  []dbConstraintZone
+	)
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		_, err := s.getModelUUID(ctx, tx)
+		if err != nil {
+			return errors.Errorf("checking if model exists: %w", err)
+		}
+
+		cons, err = s.getModelConstraints(ctx, tx)
+		if err != nil {
+			return errors.Capture(err)
+		}
+		err = tx.Query(ctx, selectTagStmt).GetAll(&tags)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("getting constraint tags: %w", err)
+		}
+		err = tx.Query(ctx, selectSpaceStmt).GetAll(&spaces)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("getting constraint spaces: %w", err)
+		}
+		err = tx.Query(ctx, selectZoneStmt).GetAll(&zones)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("getting constraint zones: %w", err)
+		}
```

Comment:
We can do this with left joins right? Why use several queries when one can do the job?

Reply:
This would become a big permutation problem for an explosion in more data and harder to read code.

Possibly I am missing something here?

Reply:
We'd need to measure query efficiency etc to make a definitive call. No need to do that right now.